test/extended/builds/build_pruning.go

```diff
@@ -149,4 +152,34 @@ var _ = g.Describe("[builds][pruning] prune builds based on settings in the buil
 
 	})
 
+	g.It("buildconfigs should have a default history limit set when created via the group api", func() {
```

Comment:
Seems like you might want this as an integration test to stop any merge from succeeding if it breaks this.